apps/spruce/src/App.tsx

```diff
@@ -1,43 +1,50 @@
 import * as React from "react";
-import {
-  Route,
-  createBrowserRouter,
-  createRoutesFromElements,
-  RouterProvider,
-} from "react-router-dom";
+import { createBrowserRouter, Outlet, RouterProvider } from "react-router-dom";
+import ProtectedRoute from "@evg-ui/lib/components/ProtectedRoute";
+import { AuthProvider } from "@evg-ui/lib/context/AuthProvider";
+import LoginPage from "@evg-ui/lib/pages/LoginPage";
 import { Content } from "components/Content";
 import { ErrorBoundary } from "components/ErrorHandling";
 import { GlobalStyles } from "components/styles";
 import { routes } from "constants/routes";
-import { ContextProviders } from "context/Providers";
-import GQLWrapper from "gql/GQLWrapper";
-import { Login } from "pages/Login";
-import { isDevelopmentBuild, isLocal } from "utils/environmentVariables";
+import ContextProviders from "context/Providers";
+import { getEvergreenUrl, isLocal } from "utils/environmentVariables";
 
-const browserRouter = createBrowserRouter(
-  createRoutesFromElements(
-    <>
-      {(isDevelopmentBuild() || isLocal()) && (
-        <Route element={<Login />} path={routes.login} />
-      )}
-      <Route
-        element={
-          <GQLWrapper>
-            <Content />
-          </GQLWrapper>
-        }
-        path="/*"
-      />
-    </>,
-  ),
-);
+const router = createBrowserRouter([
+  {
+    element: (
+      <AuthProvider
+        evergreenAppURL={getEvergreenUrl()}
+        localAuthRoute={routes.login}
+        remoteAuthURL={`${getEvergreenUrl()}/login`}
```

Comment:
I'm confused how this would work in the CorpSecure case.

Reply:
So evergreen already handles the corp secure login portion with [this login/redirect logic](https://github.com/evergreen-ci/evergreen/blob/ceae5aa100d628053b09b71a6ab4f80aabab4e48/service/ui.go#L290).  Instead of inventing our own I ended up switching to just using Evergreens existing implementation. Its been way more reliable in setting the necessary cookies then sending then relying on a UI only redirect solution.

Reply:
It also does a bunch of [other validation steps](https://github.com/evergreen-ci/gimlet/blob/622a429ac45676bc24b5caa3ff6d086c15226f31/okta/okta.go#L431) that aren't possible with just the regular okta redirect flow we were doing before.

Reply:
but isn't that the redirect to Evergreen's Okta auth? (rather than Corpsecure's)